pkg/k8shandler/statefulset.go

```diff
@@ -507,3 +507,20 @@ func (node *statefulSetNode) isChanged() bool {
 
 	return changed
 }
+
+func (node *statefulSetNode) updateStuckPod(upgradeStatus *api.ElasticsearchNodeStatus) error {
+	if node.isChanged() {
+		node.setReplicaCount(0)
+		defer node.setReplicaCount(*node.self.Spec.Replicas)
```

Comment:
this is cool, but what if we return due to a failure. we don't want to scale back up then. i think we would want to ensure that the `RetryOnConflict` was successful and then scale up.